### PR TITLE
Remove redundant `throws` declarations

### PR DESCRIPTION
**Pull request by stefanvodita**
I've noticed unused `throws` declarations multiple times and thought I'd see if they could be removed. This is the result of running IntelliJ's code inspection.
There is an argument to be made that not all of these should be removed because a user could have extended one of the classes and relies on a method throwing a certain exception. Even if that's the case, there's still some `throws` declarations we can remove, e.g. in tests.

### CODE REVIEW

**Comment by stefanvodita** [comment]
It looks like the automatic checker caught some more unused declared exceptions. Not sure why I didn’t catch these when running `./gradlew check`. I’ll investigate.

**Comment by mkhludnev** [comment]
Won't it change signature in an incompatible way?

**Comment by stefanvodita** [comment]
It would. I think there’s three categories of methods as far as this change is concerned:
1. Methods the user can't override. We can safely change all of these.
2. Methods the user can override.
    1. Those where the user should not override and throw the exception. We can change these in the next major version.
    2. Those where the user is expected to override and throw the exception. We have to keep them.

I don't know if cases 2.i. and 2.ii. actually exist. For now, I can try to take care of case 1.

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/Word2VecModel.java [hunk 1]
```diff
@@ -88,7 +87,7 @@ public int size() {
   }
 
   @Override
-  public RandomAccessVectorValues<float[]> copy() throws IOException {
+  public RandomAccessVectorValues<float[]> copy() {
```
Comment (benwtrent) [inline lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/Word2VecModel.java:90]: This is an example of one you cannot remove. The class is `public` and not final. So somebody could override the class and thus override this method.
`  exports org.apache.lucene.analysis.synonym.word2vec;` also indicates its exported in the public API for this Apache Lucene.

**Review by benwtrent — CHANGES_REQUESTED** [review]
While we should strive for tighter linting controls for certain things. I am not sure this is one of them.

Any method that is not `private`, `final` part of a `final` class or part of a class that is not exported shouldn't have its exception declaration changed.

Additionally, sub-classes of any public API classes could be taking advantage of the exception declaration, even if the parent class is not. So, I don't think we could even do this safely as a "breaking change".

**Comment by stefanvodita** [comment]
Those are good points. What if we took a more targeted approach? Maybe we can remove unused `throws` from the tests as a step in the right direction. I’ve pushed a commit doing that. It’s a large diff, so beware, but I think it’s doing the right thing.

**Comment by uschindler** [comment]
In general I would like to keep all tests with "throws Expection". This makes creating/modifying tests faster. I use a template whenever I create a new test.

So please keep the tests as is. Thanks.

**Comment by stefanvodita** [comment]
Thanks for the feedback Uwe. I thought changing the tests would be uncontroversial, but it sounds like you have a use-case for those exceptions. I’ll close this PR.

**Pull request closed without merging**